<?xml version="1.0"?>
<launch>
	<!--********** arg **********-->
	<arg name="mav_name" default="iris"/>
	<arg name="drone_name" default="$(arg mav_name)"/>
	<arg name="roboID" default="0"/>

	<arg name="is_input_joystick" default="false"/>
	<arg name="enable_meshes" default="true"/>
	<arg name="enable_wind" default="true"/>
	<arg name="enable_physics" default="true"/>
	<arg name="enable_sensors" default="true"/>
	<arg name="enable_logging" default="false" />
	<arg name="enable_ground_truth" default="true" />
	<arg name="enable_mavlink_interface" default="false"/>

	<arg name="world_name" default="basic"/>
	<arg name="debug" default="false"/>
	<arg name="gui" default="true"/>
	<arg name="paused" default="false"/>
	<arg name="log_file" default="$(arg mav_name)_$(arg drone_name)"/>
	<arg name="wait_to_record_bag" default="false"/>
	<arg name="verbose" default="true"/>

	<arg name="X" default="0.0"/>
	<arg name="Y" default="0.0"/>
	<arg name="Z" default="0.0"/> -->

		<!--********** load env **********-->
		<env name="GAZEBO_MODEL_PATH" value="${GAZEBO_MODEL_PATH}:$(find rotors_gazebo)/models"/>
		<env name="GAZEBO_RESOURCE_PATH" value="${GAZEBO_RESOURCE_PATH}:$(find rotors_gazebo)/models"/>

		<!--********** spawn model **********-->
		<include file="$(find rotors_gazebo)/launch/spawn_mav.launch">
			<arg name="mav_name" value="$(arg mav_name)" />
			<!-- odom and imu should be included in mav_generic_odometry_sensor.gazebo -->
			<arg name="model" value="$(find rotors_description)/urdf/mav_generic_odometry_sensor.gazebo" />
			<arg name="enable_logging" value="$(arg enable_logging)" />
			<arg name="enable_ground_truth" value="$(arg enable_ground_truth)" />
			<arg name="log_file" value="$(arg log_file)"/>
			<arg name="namespace" value="$(arg drone_name)"/>
			<arg name="x" value="$(arg X)"/>
  			<arg name="y" value="$(arg Y)"/>
  			<arg name="z" value="$(arg Z)"/>
		</include>

		<node name="roll_pitch_yawrate_thrust_controller_node" pkg="rotors_control" type="roll_pitch_yawrate_thrust_controller_node" output="screen">
			<rosparam command="load" file="$(find rotors_gazebo)/resource/roll_pitch_yawrate_thrust_controller_$(arg mav_name).yaml" />
			<rosparam command="load" file="$(find rotors_gazebo)/resource/$(arg mav_name).yaml" />
			<remap from="odometry" to="odometry_sensor1/odometry" />
		</node> 

		<node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />
		<node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" />

</launch>


